### PR TITLE
Redis Session Support

### PR DESCRIPTION
**Pull request by willronchetti**
- Allow the building of the server with a `redis` component
- Add a `redis` handler in the application registry
- Refactor authentication flow to introduce Redis Sessions
  - Move JWT token generation into the back-end via `/callback`
  - Generate a Redis Session Token mapped to the JWT in Redis
  - Return the session token as `jwtToken` cookie
  - Implement `LoginSuccessView`, dispatched on `/callback` that handles the UI interaction post-successful login
  - TODO: style the redirect page
  - TODO: registration

### CODE REVIEW

File: src/encoded/static/components/navigation/components/LoginSuccessView.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    
+    constructor(props) {
+        super(props);
+        this.state = {
+            readyToRedirect: false
+        };
+    }
+
+    /**
+     * This component is meant to be loaded upon navigating the /callback
```
Comment (dmichaels-harvard) [inline src/encoded/static/components/navigation/components/LoginSuccessView.js:37]: suggestion: "upon navigating from the backend (on authentication) back to the /callback"

**Review by dmichaels-harvard — APPROVED** [review]

File: src/encoded/authentication.py [hunk 2]
```diff
@@ -407,16 +433,15 @@ def login(context, request):
...
-    The front-end handles logging out by discarding the locally-held JWT from
-    browser cookies and re-requesting the current 4DN URL.
+    This endpoint is called by the front-end upon executing a logout. It will delete
+    the session token passed to it from Redis storage, which has the effect of logging
+    out the user internally. If this isn't done there is no harm as the session token
+    will expire after 3 hours anyway. 
+
+    We do not send the logout signal to Auth0 at this time, since we handle it on our end.
+    It may be undesirable to send this signal as it will log the user out of all sessions
+    tied to that account.
```
Comment (dmichaels-harvard) [inline src/encoded/authentication.py:443]: Yeah, that (undesirable - log the user out of all sessions tied to that account) sounds right.

**Review by dmichaels-harvard — APPROVED** [review]

**Review by Bianca-Morris — COMMENTED** [review]
Can you deploy this somewhere? I've had a bunch of issues running this locally, and don't have a bunch of time for debugging these days.

**Comment by willronchetti** [comment]
Closing in favor of a new PR as this one has grown stale, see py311_redis

**Pull request closed without merging**